src/tools/ScenarioMeasurement/Util/ProcessHelper.cs

```diff
@@ -36,19 +36,89 @@ public class ProcessHelper
 
         public bool RootAccess { get; set; } = false;
 
+        public bool IsRunWithExit {get;set;} = true;
+
         public ProcessHelper(Logger logger)
         {
             this.Logger = logger;
         }
 
+
+        public (Process Proc, Result Result, int Pid) Run()
+        {
+            (Result Result, int Pid) exitResult = default((Result Result, int Pid));
+            exitResult.Pid = -1;
+            Process p = null;
+            if(IsRunWithExit)
+            {
+                exitResult = RunWithExit();
+            }
+            else
+            {
+                p = RunWithNoExit();
+            }
+            return (p, exitResult.Result, exitResult.Pid);
+        }
+
+        /// <summary>
+        /// Runs the specified process and does not wait for it to exit.
+        /// </summary>
+        /// <param name="appExe">Full path to executable</param>
+        /// <param name="appArgs">Optional arguments</param>
+        /// <param name="workingDirectory">Optional working directory (defaults to current directory)</param>
+        /// <returns></returns>
+        public Process RunWithNoExit()
```

Comment:
I understand you moved the handling of stderr/stdout out of this method so you can search it instead, but it feels like a weird break of the contract for `ProcessHelper` to not handle those in the case you want to let the process run. I'd like to see there be either a default handling here that can be overridden, or a rework of the handling generally in this type that lets you insert text processing into the chain.

Reply:
So I can imagine how we could do it either way, but I have a question. The way I had imagined this method being used, basically required the use of some kind of text processing on the process to understand the current state. In all other places we are either using a GUI app, and allow for a timer to elapse, or some process that we expect to shutdown of it's own volition. But here I would posit that if you are just relying on the default output processing that you will be holding it wrong, and almost certainly not doing what you want to do. Mainly because `RunWithoutExit` gives up on handling control flow for you, where `RunWithExit` does. Maybe the right thing do here is to change the ProcessHelper type to an interface, and have a "managed" and "unmanaged" process execution. That way you at least know what you are signing up for when you use the `RunWithoutExit` mode.